proposed/2021/TransitiveDependenciesInPMUI.md

```diff
@@ -0,0 +1,95 @@
+
+### Transitive Packages details pane
+
+The Details pane will be populated for transitive packages in the same way that it is populated for other installed packages. The same mechanism will be reused for transitive packages so the user package metadata, select a version, and click the Install button to install the package which will promote it to a top-level package. There should not need to be many changes to support adding details for transitive packages.
+
+### A/B Experimentation
+
+**The Transitive Dependencies feature will be rolled out to customers as an A/B Experiment.**
+
+This document is not intended to outline the full details of the A/B Experiment but to highlight some key points about releasing the feature as an experiment.
+
+Running an A/B Experiment with the transitive dependencies feature will allow us to take a data-driven approach to deciding if the feature we built sufficently addresses the desired goal or if we need further investment to positively impact the user experience.
+
+It also allows us the ability to turn off the experiment remotely if we find significant regressions in the experience or remotely broaden the audience if the data shows a positive impact.
+
+We will make use of existing engagement metrics to determine the feature's impact on the user's interaction with the PMUI. Namely, we will look closely at NuGet Actions (Package installs, updates, and uninstalls), as well as PMUI Refresh events on each of the tabs. We will consider adding additional metrics specific to transitive dependencies in consultation with the VS Experimentation team.
+
+
+## Considerations
+
+* The transitive packages list should be collapsed by default because the user may not need to see transitive package information every time they view the Installed tab. There is a potential opportunity to delay processing or querying transitive dependency or the top-level packages that installed the transitive dependencies until after the list is expanded. However, doing so would be premature optimization so this will only be considered if it is shown to have a significant impact on performance. Since the assets file is read from disk into memory, and processing will occur on that in-memory representation, I do not anticipate a significant impact, but this consideration can be visited as an option if we see otherwise.
+* Transitive dependencies and the top-level packages that caused them to be installed are to be read from the assets file on disk. If that file is not present when the PMUI is loaded and installed packages are displayed, we will need to understand the impact it will have on this scenario and devise a solution. It is difficult to anticipate what the best options are without reaching that point, but some options may include doing a restore to create the file, prompting the user, or not displaying transitive dependencies and informing the user why they are not visible until after a restore is completed and the assets file is created.
```

Comment:
> or not displaying transitive dependencies and informing the user why they are not visible until after a restore is completed and the assets file is created.

This is my vote. The UI should immediately handle the situation where they aren't available. Not wait on anything. The UI today blocks itself on too many background tasks. Any new UI should strongly consider relaying state information to the user in a friendly way.

Reply:
Thanks for your input! This sounds good to me too. Maybe we'd show a message in the place where transitive dependencies would appear or somewhere at the top of the UI. We can figure out a good place for it once we find if we need to solve this scenario and how.